docs/_guide/events.md

```diff
@@ -8,96 +8,156 @@ slug: events
+*   Adding the event listener in `connectedCallback` (instead of, for example, the constructor or `firstUpdated`) ensures that your component will re-create its event listener if it is disconnected and subsequently reconnected to DOM.
 
-    If your component adds an event listener to anything except itself or its children–for example, to `Window`, `Document`, or some element in the main DOM–you should add the listener in `connectedCallback` and remove it in `disconnectedCallback`.
-    
-    *   Removing the event listener in `disconnectedCallback` ensures that any memory allocated by your component will be cleaned up when your component is destroyed or disconnected from the page. 
+**Example**
 
-    *   Adding the event listener in `connectedCallback` (instead of, for example, the constructor or `firstRendered`) ensures that your component will re-create its event listener if it is disconnected and subsequently reconnected to DOM.
-    
-    **Example**
+```js
+connectedCallback() {
+  super.connectedCallback();
+  window.addEventListener('resize', this.handleResize);
+}
+disconnectedCallback() {
+  window.removeEventListener('resize', this.handleResize);
+  super.disconnectedCallback();
+}
+```
+
+### Add event listeners after first paint
+
+Sometimes, you may want to defer adding an event listener until after first paint—for example, if you're adding a lot of listeners and first paint performance is critical.
+
+LitElement doesn't have a specific lifecycle callback called after first paint, but you can use this pattern with the `firstUpdated` lifecycle callback:
+
+```js
+async firstUpdated() {
+  // Give the browser a chance to paint
+  await new Promise((r) => setTimeout(r, 0));
+  this.addEventListener('click', this._handleClick);
+}
+```
+
+`firstUpdated` fires after the first time your component has been updated and called its `render` method, but **before the browser has had a chance to paint**. The `Promise`/`setTimeout` line yields to the browser
     
-    ```js
-    connectedCallback() {
-      super.connectedCallback();
-      document.addEventListener('readystatechange', this.handleChange);
-    }
-    disconnectedCallback() {
-      document.removeEventListener('readystatechange', this.handleChange);
-      super.disconnectedCallback();
-    }
-    ```
+See [firstUpdated](/guide/lifecycle#firstupdated) in the Lifecycle documentation for more information.
+
 
-### Using `this` in event handlers
+## Using `this` in event listeners
 
-The default JavaScript context object (`this`) inside an event handler belonging to a LitElement-based component is the component itself. 
+Event listeners added using the declarative (`@event`) syntax in the template are automatically _bound_ to the component.
```

Comment:
You should probably keep the "bound" verbiage here, but technically we don't bind the function, we just call it with the right `this` value. The difference is that we don't create a new bound function for each event binding * element instance, but just reuse the prototype type method. It's only a thing because people often criticize React for creating so many closures.

Reply:
OK, so just for the record--I'm _not_ changing any doc here, right?